tests/test_ethereum_payment_provider.py

```diff
@@ -0,0 +1,137 @@
+import datetime
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import timezone
+from django.test import RequestFactory
+from django.contrib.sessions.backends.db import SessionStore
+
+import pytest
+
+from pretix.base.models import (
+    Event,
+    Organizer,
+)
+
+from pretix_eth.payment import Ethereum
+
+
+ETH_ADDRESS = '0xeee0123400000000000000000000000000000000'
+DAI_ADDRESS = '0xda10123400000000000000000000000000000000'
+
+
+@pytest.fixture
+def event(transactional_db):
+    now = timezone.now()
+    presale_start_at = now + datetime.timedelta(days=2)
+    presale_end_at = now + datetime.timedelta(days=6)
+    start_at = now + datetime.timedelta(days=7)
+    end_at = now + datetime.timedelta(days=14)
+
+    organizer = Organizer.objects.create(name='Ethereum Foundation')
+    event = Event.objects.create(
+        name='Devcon',
+        organizer=organizer,
+        date_from=start_at,
+        date_to=end_at,
+        presale_start=presale_start_at,
+        presale_end=presale_end_at,
+        location='Osaka',
+    )
+    return event
+
+
+@pytest.fixture
+def provider(event):
+    provider = Ethereum(event)
+    return provider
+
+
+@pytest.mark.django_db
+def test_provider_properties(provider):
+    assert provider.identifier == 'ethereum'
+    assert provider.verbose_name == 'Ethereum'
+    assert provider.public_name == 'Ethereum'
```

Comment:
What is a `provider` in this context? Should this be "Ether" instead of "Ethereum"? Or maybe it should be the "Ethereum Foundation"? Or "Ethereum-based"?

Reply:
payment provider.  I'll get the class name updated to something less utterly generic.

Reply:
Cool. So this text (`verbose_name` and `public_name`) is the text header where you choose between Bitcoin/CC/Ethereum payments? I guess `Ethereum-based` is the most accurate, since it includes DAI. No strong opinions though.